### PR TITLE
big improvements to readme

### PR DESCRIPTION
**Pull request by epwalsh**
While the situation is improving, model discoverability is still not great. This adds a lot of info to the README to help with that.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +26,100 @@
...
+## About
+
+This repository contains the components - such as [`DatasetReader`](https://docs.allennlp.org/main/api/data/dataset_readers/dataset_reader/#datasetreader), [`Model`](https://docs.allennlp.org/main/api/models/model/#model), and [`Predictor`](https://docs.allennlp.org/main/api/predictors/predictor/#predictor) classes - for applying [AllenNLP](https://github.com/allenai/allennlp) to a wide variety of NLP [tasks](#tasks-and-components).
+It also provides an easy way to download and use [pre-trained models](#pre-trained-models) that were trained with these components.
+
+### Tasks and components
+
+This is an overview of the tasks supported by the AllenNLP Models library along with the corresponding components provided, organized by category. For a more comprehensive overview, see the [AllenNLP Models documentation](https://docs.allennlp.org/models/main/) or the [Paperswithcode page](https://paperswithcode.com/lib/allennlp).
+
+- [**Classification**](https://github.com/allenai/allennlp-models/tree/main/allennlp_models/classification)
```
Comment (ArjunSubramonian) [inline README.md:38]: Perhaps add a description of what Classification is, like you've done for the other categories, e.g. prediction of discrete integer labels for examples

File: README.md [hunk 2]
```diff
@@ -26,6 +26,100 @@
...
+  
+  Classification is a broad category that contains many more specific tasks such as Sentiment Analysis and Binary Question Answering.
+
+  🛠 **Components provided:** Dataset readers for various datasets, including [BoolQ](https://docs.allennlp.org/models/main/models/classification/dataset_readers/boolq/) and [SST](https://docs.allennlp.org/models/main/models/classification/dataset_readers/stanford_sentiment_tree_bank/), as well as a [Biattentive Classification Network](https://docs.allennlp.org/models/main/models/classification/models/biattentive_classification_network/) model.
+
+- [**Coreference Resolution**](https://github.com/allenai/allennlp-models/tree/main/allennlp_models/coref)
+
+  Coreference resolution is the task of finding all expressions that refer to the same entity in a text. It is an important step for many higher level NLP tasks that involve natural language understanding such as document summarization, question answering, and information extraction.
```
Comment (ArjunSubramonian) [inline README.md:46]: Should we cite https://nlp.stanford.edu/projects/coref.shtml here?

File: README.md [hunk 3]
```diff
@@ -26,6 +26,100 @@
...
+
+- [**Language Modeling**](https://github.com/allenai/allennlp-models/tree/main/allennlp_models/lm)
+
+  Language Modeling tasks involve learning a probability distribution over sequences of tokens.
+
+  🛠 **Components provided:** Several language model implementations, such as a [Masked LM](https://docs.allennlp.org/models/main/models/lm/models/masked_language_model/) and a [Next Token LM](https://docs.allennlp.org/models/main/models/lm/models/next_token_lm/).
+
+- [**Multiple Choice**](https://github.com/allenai/allennlp-models/tree/main/allennlp_models/mc)
+
+  Multiple Choice tasks are tasks that involve selecting a correct answer out of a set of possible answers, given some input.
```
Comment (ArjunSubramonian) [inline README.md:64]: Personally confused on the distinction between `multiple choice` and `classification`

**Review by dirkgr — DISMISSED** [review]
Big fan!

File: README.md (hunk 3 above)
Comment (dirkgr) [reply README.md:64]: Multiple choice is choosing between alternatives. Classification is evaluating each alternative on its own. In practice we can get human-level performance by classifying each option on its own, but I suspect that's a problem with the datasets. I wonder if a model that can actually compare possible answers against each other could do even better than what we do now.

Comment (epwalsh) [reply README.md]: Just updated. Let me know if that makes sense.

File: README.md (hunk 2 above)
Comment (epwalsh) [reply README.md]: Thanks, added!

File: README.md (hunk 1 above)
Comment (epwalsh) [reply README.md:38]: Yeup, good call. I was being lazy 😜

**Review by ArjunSubramonian — APPROVED** [review]
Looks good to me!

**Pull request merged**